app/jobs/v3/delete_route_binding_job.rb

```diff
@@ -47,24 +49,43 @@ def perform
         if @first_time
           @first_time = false
           delete_result = action.delete(binding, async_allowed: true)
-          return finish if delete_result.is_a? V3::ServiceRouteBindingDelete::DeleteComplete
+          if delete_result[:finished]
+            finish
+          end
         end
 
         polling_status = action.poll(binding)
-        case polling_status
-        when V3::ServiceRouteBindingDelete::DeleteComplete
+        if polling_status[:finished]
           finish
-        when V3::ServiceRouteBindingDelete::DeleteInProgress
-          unless polling_status.retry_after.nil?
-            self.polling_interval_seconds = polling_status.retry_after.to_i
-          end
+        end
+
+        if polling_status[:retry_after].present?
+          self.polling_interval_seconds = polling_status[:retry_after]
         end
       rescue => e
+        if route_binding.reload.last_operation.state != 'failed'
```

Comment:
You could technically skip this and always save "state failed" 🤔

Reply:
the intention here was to not change the description - as set by the previous steps. Happy to change it if you think it makes sense.

E.g poll can through LastOperationFailed and also sets the state to failed with the right description.

Reply:
Fair enough. I was assuming that whatever is raising, is doing it with the correct error message. But maybe not?